fe/fe-core/src/main/java/org/apache/doris/rewrite/OrToInRule.java

```diff
@@ -0,0 +1,109 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.rewrite;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.doris.analysis.Analyzer;
+import org.apache.doris.analysis.BinaryPredicate;
+import org.apache.doris.analysis.CompoundPredicate;
+import org.apache.doris.analysis.Expr;
+import org.apache.doris.analysis.InPredicate;
+import org.apache.doris.analysis.LiteralExpr;
+import org.apache.doris.analysis.SlotRef;
+import org.apache.doris.common.AnalysisException;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * OR Expr cannot be pushdown on the storage side, rewrite it to InPredicate.
+ * SQL: select * from tbl where k = 0 or k = 1 or k in (2, 3);
+ * After rewrite: select * from tbl where k in (0, 1, 2, 3)
+ * */
+public class OrToInRule implements ExprRewriteRule {
```

Comment:
Hi~ I think we can add rewrite rule in the same java file, because they both belong to Predicate 'OR' rewrite.
fe/fe-core/src/main/java/org/apache/doris/rewrite/CompoundPredicateWriteRule.java
related pr: https://github.com/apache/incubator-doris/pull/6518